### PR TITLE
Use api snippets for auth-service interfaces

### PR DESCRIPTION
**Pull request by kkzeng**
- Remove old files
- Replace all references with api-snippet references
- Change buildopen:api script to resolve api-snippet references and validate new api.yaml

### CODE REVIEW

**Review by lewisdaly — COMMENTED** [review]
Awesome! Thanks Kenneth. 

Btw since we changed the file name from `swagger.json`, we should update the [documentation here](https://github.com/mojaloop/auth-service#updating-the-openapi-swagger-spec)

**Comment by kleyow** [comment]
looks good to me

File: src/server/handlers/metrics.ts [hunk 1]
```diff
@@ -26,7 +26,11 @@
 import Metrics from '@mojaloop/central-services-metrics'
 import { Request, ResponseToolkit, ResponseObject } from '@hapi/hapi'
 
-export function get (_: Request, h: ResponseToolkit): ResponseObject {
+function get (_: Request, h: ResponseToolkit): ResponseObject {
```
Comment (eoln) [inline src/server/handlers/metrics.ts:29]: I would keep export here to have the same consistency wich we keep in other handlers

File: src/shared/histogram.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+export type THandlerFunc = (request: Request, h: ResponseToolkit) => Promise<ResponseObject>
+
+/**
+ * @function wrapWithHistogram
+ * @description Wraps a handler function with a histogram of the given name
+ * @param {THandlerFunc} handler The handler function to be wrapped
+ * @param {[string, string, Array<string>]} histogramParams The params of the histogram
+ */
+function wrapWithHistogram (handler: THandlerFunc, histogramParams: [string, string, string[]]): THandlerFunc {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (eoln) [inline src/shared/histogram.ts:14]: where do we have `any` here? drop the linting exception

**Review by eoln — CHANGES_REQUESTED** [review]
really small things to fix

**Comment by kkzeng** [comment]
Woohoo, finally CI is green :D Time to address comments

File: src/shared/histogram.ts (hunk 2 above)
Comment (kkzeng) [reply src/shared/histogram.ts]: Sure, I can remove it. Not sure why it's there either.
I was wondering if I need the histogram for auth-service?
I removed the histogram in my latest commit but I can add it back

File: src/server/handlers/metrics.ts (hunk 1 above)
Comment (kkzeng) [reply src/server/handlers/metrics.ts]: yup agreed

File: src/server/handlers/metrics.ts [hunk 3]
```diff
@@ -26,7 +26,11 @@
 import Metrics from '@mojaloop/central-services-metrics'
 import { Request, ResponseToolkit, ResponseObject } from '@hapi/hapi'
 
-export function get (_: Request, h: ResponseToolkit): ResponseObject {
+function get (_context: any, _: Request, h: ResponseToolkit): ResponseObject {
```
Comment (eoln) [inline src/server/handlers/metrics.ts:29]: use here Context interface you defined in`~/server/plugins`

**Review by eoln — CHANGES_REQUESTED** [review]

File: src/server/handlers/metrics.ts (hunk 3 above)
Comment (kkzeng) [reply src/server/handlers/metrics.ts:29]: yes, I think I didn't save in VSCode, I fixed it now. thanks for catching this

**Review by lewisdaly — APPROVED** [review]
Looks good! Thanks @kkzeng

**Review by eoln — APPROVED** [review]
lgtm

**Pull request merged**